### ISSUE TITLE
Why 'get_provisioning_profile' downloads two profiles instead one?

### ISSUE THREAD

**Issue by KomarovEV** [fastlane/fastlane#14467]
### Question Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

##### What tool/action do you have a question about?

'get_provisioning_profile' && 'sigh fetch_profiles'

##### Is this a question about documentation?

No.

##### Is this a question about a third party plugin? (If so, please go to the plugin repository first)

No.

#### Question Description

There is a command 'get_provisioning_profile' which used by us like this: 

```
    get_provisioning_profile(
      output_path: ENV["FL_PROVISIONING_PROFILES"],
      development: true,
      app_identifier: ENV["APP_IDENTIFIER_DEVELOP"]
    )
```

It founds one, but downloads two identical provisioning profiles with different names. One of it has typical provisioning profile name, like 'c4f37a45-a234-4123-b740-7601bfc8b800.mobileprovision', but the another one named "Development_app.id.name.mobileprovision". 

Output:

```
Found 1 matching profile(s)
Downloading provisioning profile...
Successfully downloaded provisioning profile...
Installing provisioning profile...
/Users/User/Library/MobileDevice/Provisioning Profiles/Development_app.id.name.mobileprovision.mobileprovision
```

Then it tries to use the second one:

```
 ▸ ❌  error: No profiles for 'app.id.name' were found: Xcode couldn't find any iOS App Development provisioning profiles matching 'app.id.name'. Automatic signing is disabled and unable to generate a profile. To enable automatic signing, pass -allowProvisioningUpdates to xcodebuild. (in target 'AppTarget')

** ARCHIVE FAILED **
Exit status: 65

```

However, the following command does the thing and successfully downloads provisioning profile we are needed (with 'c4f37a45-a234-4123-b740-7601bfc8b800.mobileprovision'-like name), then executes successfully:

```
   sh "sigh -a #{ENV["APP_IDENTIFIER_DEVELOP"]} fetch_profiles --development -output_path #{ENV["FL_PROVISIONING_PROFILES"]}"
```

#### Expected behavior

Booth commands do the same thing - downloads and uses only one provisioning profile with correct name and exits without errors.

What is the difference and why 'get_provisioning_profile' does not work as expected?

<details><summary>✅ fastlane environment ✅</summary>

### Stack

| Key                         | Value                                       |
| --------------------------- | ------------------------------------------- |
| OS                          | 10.13.6                                     |
| Ruby                        | 2.2.4                                       |
| Bundler?                    | false                                       |
| Git                         | git version 2.7.0                           |
| Installation Source         | ~/.fastlane/bin/bundle/bin/fastlane         |
| Host                        | Mac OS X 10.13.6 (17G5019)                  |
| Ruby Lib Dir                | ~/.fastlane/bin/bundle/lib                  |
| OpenSSL Version             | OpenSSL 1.0.2g  1 Mar 2016                  |
| Is contained                | false                                       |
| Is homebrew                 | true                                        |
| Is installed via Fabric.app | false                                       |
| Xcode Path                  | /Applications/Xcode.app/Contents/Developer/ |
| Xcode Version               | 10.1                                        |

### System Locale

| Variable | Value       |   |
| -------- | ----------- | - |
| LANG     | en_US.UTF-8 | ✅ |
| LC_ALL   | en_US.UTF-8 | ✅ |
| LANGUAGE | en_US.UTF-8 | ✅ |

### fastlane gems

| Gem      | Version | Update-Status |
| -------- | ------- | ------------- |
| fastlane | 2.118.1 | ✅ Up-To-Date  |

### Loaded fastlane plugins:

**No plugins Loaded**

<details><summary><b>Loaded gems</b></summary>

| Gem                       | Version      |
| ------------------------- | ------------ |
| slack-notifier            | 2.3.2        |
| atomos                    | 0.1.3        |
| CFPropertyList            | 2.3.5        |
| claide                    | 1.0.2        |
| colored2                  | 3.1.2        |
| nanaimo                   | 0.2.6        |
| xcodeproj                 | 1.8.1        |
| rouge                     | 2.0.7        |
| xcpretty                  | 0.3.0        |
| terminal-notifier         | 2.0.0        |
| unicode-display_width     | 1.1.3        |
| terminal-table            | 1.7.3        |
| plist                     | 3.2.0        |
| public_suffix             | 2.0.5        |
| addressable               | 2.5.1        |
| multipart-post            | 2.0.0        |
| word_wrap                 | 1.0.0        |
| tty-screen                | 0.6.4        |
| tty-cursor                | 0.5.0        |
| tty-spinner               | 0.8.0        |
| babosa                    | 1.0.2        |
| colored                   | 1.2          |
| highline                  | 1.7.8        |
| commander-fastlane        | 4.4.6        |
| excon                     | 0.55.0       |
| faraday                   | 0.12.1       |
| unf_ext                   | 0.0.7.4      |
| unf                       | 0.1.4        |
| domain_name               | 0.5.20170404 |
| http-cookie               | 1.0.3        |
| faraday-cookie_jar        | 0.0.6        |
| fastimage                 | 2.1.0        |
| gh_inspector              | 1.1.3        |
| json                      | 1.8.1        |
| mini_magick               | 4.5.1        |
| multi_json                | 1.12.1       |
| multi_xml                 | 0.6.0        |
| rubyzip                   | 1.2.2        |
| security                  | 0.1.3        |
| xcpretty-travis-formatter | 0.0.4        |
| dotenv                    | 2.2.0        |
| bundler                   | 1.14.6       |
| faraday_middleware        | 0.11.0.1     |
| naturally                 | 2.2.0        |
| simctl                    | 1.6.5        |
| uber                      | 0.0.15       |
| declarative               | 0.0.10       |
| declarative-option        | 0.1.0        |
| representable             | 3.0.4        |
| retriable                 | 2.1.0        |
| mime-types-data           | 3.2016.0521  |
| mime-types                | 3.1          |
| jwt                       | 1.5.6        |
| signet                    | 0.11.0       |
| memoist                   | 0.16.0       |
| os                        | 0.9.6        |
| googleauth                | 0.6.7        |
| httpclient                | 2.8.3        |
| google-api-client         | 0.23.9       |
| google-cloud-env          | 1.0.5        |
| google-cloud-core         | 1.2.7        |
| digest-crc                | 0.4.1        |
| google-cloud-storage      | 1.15.0       |
| emoji_regex               | 0.1.1        |
| libxml-ruby               | 3.0.0        |
| psych                     | 2.0.8        |
</details>

*generated on:* **2019-03-25**
</details>

**Comment by joshdholtz** [comment]
@KomarovEV Thanks for asking this question! I do agree that it seems like these should do the same thing? 🤔 I will definitely look into this because now I am really curious 😬

**Comment by KomarovEV** [comment]
Ok, we found an issue in 

```
   sh "sigh -a #{ENV["APP_IDENTIFIER_DEVELOP"]} fetch_profiles --development -output_path #{ENV["FL_PROVISIONING_PROFILES"]}"

```

It should be ...-o #{ENV["FL_PROVISIONING_PROFILES"]}"

But there is still couple questions in this situation.

1. If we use ```sigh download_all --download_xcode_profiles``` we got two identical files in the output folder, but one with ```Development_``` prefix. Why there are two files downloaded?

2. If we use ```get_provisioning_profile``` it doesn't download iOS Team Profile and we got an error describing above (Exit status: 65). How to solve that issue?

3. If we use ```sigh download_all --download_xcode_profiles``` we unable to specify app id for which we need to fetch profiles. Is there is a way to specify it?

Thank you.
